amazon-kinesis-client/src/test/java/software/amazon/kinesis/retrieval/polling/PrefetchRecordsPublisherIntegrationTest.java

```diff
@@ -224,6 +227,25 @@ public DataFetcherResult answer(final InvocationOnMock invocationOnMock) throws
         verify(dataFetcher).restartIterator();
     }
 
+    @Test
+    public void testExpiredIteratorExceptionWithInnerRestartIteratorException() {
```

Comment:
How different this unit test case is from testExpiredIteratorExceptionWithIllegalStateException? I see datafetcher.getRecords is just a inner call in getRecordsRetrievalStrategy.getRecords(...)